components/layout_2020/flow/mod.rs

```diff
@@ -1013,19 +1034,35 @@ impl NonReplacedFormattingContext {
                         block_size: block_size.map(|t| t.into()),
                         style: &self.style,
                     },
+                    containing_block,
                 );
-                content_size = LogicalVec2 {
-                    inline: proposed_inline_size,
-                    block: block_size.auto_is(|| {
-                        layout
-                            .content_block_size
-                            .clamp_between_extremums(
-                                min_box_size.block.into(),
-                                max_box_size.block.map(|t| t.into()),
-                            )
-                            .into()
-                    }),
-                };
+
+                if let Some(inline_size) = layout.content_inline_size_for_table {
+                    let outer_inline_size = inline_size + pbm.padding_border_sums.inline;
+                    placement.set_inline_size(outer_inline_size, &pbm);
```

Comment:
Can this be moved to inside the if-statement right below?

Reply:
It could be moved. But setting it unconditionally is an optimization: even if the condition is false, we may need more iterations, and then we ensure that we will find a rect that is big enough. So this avoids potential useless layouts.

Reply:
Ah, right. I had forgotten that tables also need to fit among floats in the block direction! :facepalm:

Reply:
The new comment is great. Thanks!